src/lib/plans/other/AgentLoop.js

```diff
@@ -119,10 +120,39 @@ function chooseBestOption() {
 }
 
 export function updateCarriedParcelsScore() {
-    let sumScore = carriedParcels.reduce((previous, current, index) => previous + current.reward, 0)
-    let closestDelivery = findClosestDelivery()
+    let sumScore = carriedParcels.reduce((previous, current, index) => previous + current.reward, 0)   
     if (sumScore > 0){
-        console.log('updateCarriedParcelsScore', sumScore - closestDelivery.distance, sumScore, closestDelivery.distance)
-        agent.changeIntentionScore('go_deliver', [], sumScore - Math.floor(closestDelivery.distance / 2), 'go_deliver')
+        decidePickupOrDeliver(sumScore,carriedParcels);
     }
-}
\ No newline at end of file
+}
+
+function decidePickupOrDeliver(sumScore,carriedParcels){
+    const lowestValuedParcel = carriedParcels.reduce((previous, current) => previous.reward < current.reward ? previous : current, carriedParcels[0]);
+
+    const nextParcel = Array.from(parcels.values()).reduce((previous, current) => previous.reward > current.reward ? previous : current, parcels.values().next().value);
+
+    if(!nextParcel){
+        agent.push({ desire: 'go_deliver', args: [], score: sumScore, id: 'go_deliver'})
+        return;
+    }
+
+    const distanceToParcel = distance(me, nextParcel);
+    const distanceParcelToDelivery = findClosestDelivery(null, nextParcel).distance;
+  
+    const totalDistance = distanceParcelToDelivery + distanceToParcel;
+    const decayIterval = parseInt((configs.PARCEL_DECADING_INTERVAL).split('s')[0])
+    
+    const futureDeliveredReward = lowestValuedParcel.reward - (totalDistance * decayIterval);
+    
+    const futurePickUpReward = lowestValuedParcel.reward - (distanceToParcel * decayIterval);
+
+
+    if(totalDistance > futureDeliveredReward){
```

Comment:
This is a good idea, but I think we should compare:
1. the reward for ALL carried parcels, minus the distance * DECAY_INTERVAL * number of carried parcels (each parcel decays for each interval)
2. the (reward for the carried parcels + the best parcels) - (distance to parcel + distance to delivery) * DECAY_INTERVAL